files/disable_inactive_freeipa_users.sh

```diff
@@ -0,0 +1,130 @@
+  usage
+else
+  # Convert disable_time into an integer representing seconds since
+  # the epoch.
+  disable_deadline=$(date --date="$disable_time" +%S)
+
+  # kinit via the host's keytab.
+  kinit -k -t /etc/krb5.keytab
+
+  # Grab a list of all non-disabled FreeIPA users
+  users=$(
+    # Note that we disable the size and time limits normally in place
+    # when running ipa user-find.
+    ipa user-find --disabled=false --sizelimit=0 --timelimit=0 \
+      |
+      # --quiet means no printing unless the p command is used
+      sed --quiet 's/^\s*User login:\s*//p'
+  )
+
+  for user in $users; do
+    timestamps=$(ipa user-status "$user" \
+      |
+      # --quiet means no printing unless the p command is used
+      sed --quiet 's/\s*Last successful authentication:\s*//p')
+    # In the event that all timestamps are invalid (N/A) we don't want
+    # to disable the user since we don't know the last time he or she
+    # authenticated.
+    #
+    # TODO: After disable_time has elapsed with the "KDC:Disable Last
+    # Success" feature disabled we can start deleting users with all
+    # authentication timestamps invalid (N/A).  See #74 for more
+    # details.
+    all_timestamps_invalid=true
+    disable_user=true
+    for timestamp in $timestamps; do
+      # Do we have a valid timestamp in the format that the command
+      # ipa user-status uses (YYYYMMDDHHMMSSZ)?
+      if [[ $timestamp =~ ^([[:digit:]]{4})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})Z$ ]]; then
+        # At least one timestamp is valid.
+        all_timestamps_invalid=false
+
+        # 1. Reformat the timestamp to ISO 8601 format so the date
+        # command can understand it.
+        # 2. Use the date command to convert the ISO 8601 timestamp to
+        # seconds since the epoch.
+        timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
+        timestamp=$(date --date="$timestamp" +%s)
+
+        # If the timestamp postdates the deadline then the user is
+        # active and should not be disabled.
+        if ((timestamp > disable_deadline)); then
+          disable_user=false
+        fi
+      fi
+    done
+
+    # Now that we have analyzed the last authentication timestamps,
+    # disable users as necessary.
+    #
+    # TODO: Note also that after disable_time has elapsed with the
+    # "KDC:Disable Last Success" feature disabled we can start
+    # deleting users with all authentication timestamps invalid (N/A)
+    # and simplify this logic.  See #74 for more details.
+    if $disable_user; then
+      if ! $all_timestamps_invalid; then
+        ipa user-disable "$user"
+        echo User "$user" disabled because he or she is an inactive user.
+      else
+        echo User "$user" not disabled because all authentication timestamps were invalid.
+      fi
+    else
+      echo User "$user" not disabled because he or she is an active user.
```

Comment:
Use more inclusive wording:
```suggestion
        echo User "$user" disabled because they are an inactive user.
      else
        echo User "$user" not disabled because all authentication timestamps were invalid.
      fi
    else
      echo User "$user" not disabled because they are an active user.
```

Reply:
"They" is a plural pronoun.  How about:
```suggestion
        echo User "$user" disabled due to inactivity.
      else
        echo User "$user" not disabled because all authentication timestamps were invalid.
      fi
    else
      echo User "$user" not disabled due to sufficiently recent authentication.
```

Reply:
That works for me.

Reply:
Please see commit [43c6240](https://github.com/cisagov/ansible-role-freeipa-server/pull/73/commits/43c6240629a59d716428f7602f86189967251f81).